python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -605,3 +622,298 @@ def makeDataRefList(self, namespace):
             )
             self.refList.append(dataRef)
 
+def countMaskFromFootprint(mask, footprint, bitmask, ignoreMask):
+    """Function to count the number of pixels with a specific mask in a footprint
+    """
+    bbox = footprint.getBBox()
+    bbox.clip(mask.getBBox(afwImage.PARENT))
+    fp = afwImage.MaskU(bbox)
+    subMask = mask.Factory(mask, bbox, afwImage.PARENT)
+    afwDet.setMaskFromFootprint(fp, footprint, bitmask)
+    return numpy.logical_and((subMask.getArray() & fp.getArray()) > 0,
+                             (subMask.getArray() & ignoreMask) == 0).sum()
+
+
+class SafeClipAssembleCoaddConfig(AssembleCoaddConfig):
+    clipDetection = pexConfig.ConfigurableField(target=SourceDetectionTask,
+                                      doc="Detect sources on difference between unclipped and clipped coadd")
+    minClipFootOverlap = pexConfig.Field(
+        doc = "Minimum fractional overlap of clipped footprint with visit DETECTED to be clipped",
+        dtype = float,
+        default = 0.65
+    )
+    minClipFootOverlapSingle = pexConfig.Field(
+        doc = "Minimum fractional overlap of clipped footprint with visit DETECTED to be " \
+            "clipped when only one visit overlaps",
+        dtype = float,
+        default = 0.5
+    )
+    minClipFootOverlapDouble = pexConfig.Field(
+        doc = "Minimum fractional overlap of clipped footprints with visit DETECTED to be " \
+            "clipped when two visits overlap",
+        dtype = float,
+        default = 0.45
+    )
+    maxClipFootOverlapDouble = pexConfig.Field(
+        doc = "Maximum fractional overlap of clipped footprints with visit DETECTED when " \
+            "considering two visits",
+        dtype = float,
+        default = 0.15
+    )
+    minBigOverlap = pexConfig.Field(
+        doc = "Minimum number of pixels in footprint to use DETECTED mask from the single visits " \
+            "when labeling clipped footprints",
+        dtype = int,
+        default = 100
+    )
+
+    def setDefaults(self):
+        pexConfig.Config.setDefaults(self)
+        self.clipDetection.reEstimateBackground = False
+        self.clipDetection.returnOriginalFootprints = False
+        self.clipDetection.thresholdPolarity = "both"
+        self.clipDetection.thresholdValue = 2
+        self.clipDetection.nSigmaToGrow = 4
+        self.clipDetection.minPixels = 4
+        self.clipDetection.isotropicGrow = True
+        self.clipDetection.thresholdType = "pixel_stdev"
+        self.sigmaClip = 1.5
+        self.clipIter = 3
```

Comment:
It would be nice to know what the rationale behind these numbers is. I seem to remember Bob telling us that he arrived at them by experiment, which is fine, but we should write that down somewhere (if it's correct -- if it's not, we should record what really happened).